### ISSUE TITLE
Create a script for renaming the project

### ISSUE THREAD

**Issue by vishnuravi** [StanfordSpezi/SpeziTemplateApplication#47]
### Use Case

Creating a new project from the template application.

### Problem

Renaming the template application manually after starting a new project is cumbersome and time consuming.

### Solution

We should create a script that will replace "TemplateApplication" in the folder names, file names, and code of the template application with the name of the user's project automatically.

### Alternatives considered

https://github.com/yonaskolb/XcodeGen

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct and Contributing Guidelines

**Comment by vishnuravi** [comment]
The bash script below is able to clone the repository and rename the folders and directories, however, it fails to properly rename references within the Xcode project file itself, leading to errors:

```bash
#!/bin/bash

# Check if a new project name was provided as an argument
if [ "$#" -ne 1 ]; then
    echo "Usage: $0 NewProjectName"
    exit 1
fi

NEW_PROJECT_NAME=$1

# Cloning the repository
git clone https://github.com/StanfordSpezi/SpeziTemplateApplication.git
if [ $? -ne 0 ]; then
    echo "Error: Failed to clone the repository."
    exit 2
fi

# Change to the cloned repository
cd SpeziTemplateApplication || exit 3

# Rename directories and files
find . -type d -name '*TemplateApplication*' | while read -r DIR; do
    NEW_DIR=$(echo "$DIR" | sed "s/TemplateApplication/$NEW_PROJECT_NAME/g")
    mv "$DIR" "$NEW_DIR"
done

find . -type f -name '*TemplateApplication*' | while read -r FILE; do
    NEW_FILE=$(echo "$FILE" | sed "s/TemplateApplication/$NEW_PROJECT_NAME/g")
    mv "$FILE" "$NEW_FILE"
done

# Replacing occurrences inside files
find . -type f \( -name "*.xcodeproj" -or -name "*.pbxproj" -or -name "*.plist" -or -name "*.swift" \) | while read -r FILE; do
    sed -i '' "s/TemplateApplication/$NEW_PROJECT_NAME/g" "$FILE"
done

echo "Project has been cloned and renamed successfully."
```

A simple string replace may not be sufficient for the project file, and we may need to use a tool like `xcodegen`. Any thoughts?

**Comment by Supereg** [comment]
That's resolved as of #59 @PSchmiedmayer?

**Comment by PSchmiedmayer** [comment]
Yes, is resolved 🚀
